master/buildbot/util/__init__.py

```diff
@@ -119,14 +119,14 @@ def __hash__(self):
                 [getattr(self, name, self._None) for name in compare_attrs]
         return hash(tuple(map(str, alist)))
 
-    def __cmp__(self, them):
-        result = cmp(type(self), type(them))
-        if result:
-            return result
+    def isSameClassAndModule(self, them):
+        is_same_module = type(self).__module__ == type(them).__module__ \
+                         and self.__class__.__name__ == them.__class__.__name__
```

Comment:
What's the reason for using type(self) in one half of the line and self.__class__ in the other? I thought they did the same thing.